desc/profiles.py

```diff
@@ -810,23 +810,28 @@ class SplineProfile(_Profile):
         - `'catmull-rom'`: C1 cubic centripetal "tension" splines
     name : str
         name of the profile
+    df : array-like
```

Comment:
not crazy about adding this option, since it basically overrides method, and if we wanted to do something like this df would need to be part of params so it can be optimized as well.

For your case where you just want to spline iota with specified derivative for doing the map coordinates it might be easier to just call interpax directly

Reply:
Could you please explain this part more
> df would need to be part of params so it can be optimized as well

Reply:
Something like this may need to be done anyway: https://github.com/PlasmaControl/DESC/pull/1188#issuecomment-2291717947